adafruit_hid/keyboard.py

```diff
@@ -19,149 +19,448 @@
-RETURN = 0x28
-"""Moves the cursor to the next line. Also known as enter."""
-ESCAPE = 0x29
-"""Escape."""
-BACKSPACE = 0x2A
-"""Deletes a character to the left of the cursor."""
-TAB = 0x2B
-"""A tab character. It is whitespace."""
-SPACEBAR = 0x2C
-"""`` ``"""
-MINUS = 0x2D
-"""``-``"""
-EQUALS = 0x2E
-"""``=``"""
-LEFT_BRACKET = 0x2F
-"""``[``"""
-RIGHT_BRACKET = 0x30
-"""``]``"""
-BACKSLASH = 0x31
-"""``\``"""
-POUND = 0x32
-"""``#``"""
-SEMICOLON = 0x33
-"""``;``"""
-QUOTE = 0x34
-"""``'``"""
-GRAVE_ACCENT = 0x35
-r""":literal:`\``"""
-COMMA = 0x36
-"""``,``"""
-PERIOD = 0x37
-"""``.``"""
-FORWARD_SLASH = 0x38
-"""``/``"""
-
-DELETE = 0x4C
-"""Deletes a character to the right of the cursor."""
-
-RIGHT_ARROW = 0x4F
-"""Moves the cursor right."""
-LEFT_ARROW = 0x50
-"""Moves the cursor left."""
-DOWN_ARROW = 0x51
-"""Moves the cursor down."""
-UP_ARROW = 0x52
-"""Moves the cursor up."""
-
-LEFT_CONTROL = 0xE0
-"""Control modifier left of the spacebar."""
-LEFT_SHIFT = 0xE1
-"""Shift modifier left of the spacebar."""
-LEFT_ALT = 0xE2
-"""Alt modifier left of the spacebar."""
-LEFT_GUI = 0xE3
-"""GUI modifier (aka Windows key, Option or Meta) left of the spacebar."""
-RIGHT_CONTROL = 0xE4
-"""Control modifier right of the spacebar."""
-RIGHT_SHIFT = 0xE5
-"""Shift modifier right of the spacebar."""
-RIGHT_ALT = 0xE6
-"""Alt modifier right of the spacebar."""
-RIGHT_GUI = 0xE7
-"""GUI modifier (aka Windows key, Option or Meta) right of the spacebar."""
+from micropython import const
+import usb_hid
+
+class Keyboard:
+    """Send HID keyboard reports. You can send low-level raw keycodes, or send ASCII strings.
+
+    Constants for non-printing and a few printing keys are defined here for convenience.
+    Constants for the printing keys (a, comma, etc.) are not included to save memory space.
+    Use the unshifted ASCII character instead. So to indicate the ``A`` key, use 'a' (not 'A').
+    Likewise, for example, for the comma key, use ',', not '<'.
+
+    USB keycodes are defined in  http://www.usb.org/developers/hidpage/Hut1_12v2.pdf#page=58.
+
+    Remember that keycodes are the names for key *positions* on a US keyboard, and may
+    not correspond to the character that you mean to send if you want to emulate non-US keyboard.
+    For instance, on a French keyboard (AZERTY instead of QWERTY),
+    the keycode for 'q' is used to indicate an 'a'. Likewise, 'y' represents 'z' on
+    a German keyboard. This is historical: the idea was that the keycaps could be changed
+    without changing the keycodes sent, so that different firmware was not needed for
+    different variations of a keyboard.
+    """
+
+    ENTER = 0x28
+    """Keycode for Return (Enter)"""
+    RETURN = ENTER
+    """Alias for Enter"""
+    ESCAPE = 0x29
+    BACKSPACE = 0x2A
+    """Delete backward (Backspace)"""
+    TAB = 0x2B
+    SPACEBAR = 0x2C
+
+    CAPS_LOCK = 0x39
+    """Caps Lock"""
+
+    F1 = 0x3A
+    """Function key F1"""
+    F2 = 0x3B
+    """etc."""
```

Comment:
Originally I had everything docstring'd throughly as you did, and I couldn't even write resulting source code into 64K flash. So I thought that saying `""" Caps Lock"""` for `CAPS_LOCK` and the like was probably just redundant, and put docstrings only on the keys that were non-obvious. Ialso included some notes from the usb.org document. But I can put them back, since it seems like `.mpy` is going to be a necessity.

It is too bad that the keycodes don't necessarily match their keycaps except on a US keyboard. I was even thinking of using names like `US_A` to emphasize the point.